### PR TITLE
File tree list

### PR DESCRIPTION
**Pull request by lucianolorenti**
I have modified the files panel. Now it looks like this:
Files Panel

In this moment it is not checking if a file is being overwritten. I will add it later.

The "Add to Path" item typo is fixed.

### CODE REVIEW

**Comment by jonathanBieler** [comment]
Looks awesome, I'll just wait for the path_at_pos issue to be fixed before merging, because it doesn't work with the current Gtk package.

Also you are using two-spaces for indentation instead of four. Not a big deal, but if you can easily fix it, it would be nicer.

File: src/GtkExtra.jl [hunk 1]
```diff
@@ -0,0 +1,35 @@
+
```
Comment (jonathanBieler) [inline src/GtkExtra.jl]: We can put all this in GtkExtenstions.jl.

File: src/GtkIDE.jl [hunk 2]
```diff
@@ -70,12 +70,15 @@ include("PlotWindow.jl")
...
 include("Console.jl")
 include("Editor.jl")
+include("GtkExtra.jl")
 include("PathDisplay.jl")
 
 if sourcemap == nothing
     sourcemap = @GtkBox(:v)
 end
 
+GtkIconThemeAddResourcePath(GtkIconThemeGetDefault(), joinpath(dirname(@__FILE__),"../icons/"))
```
Comment (jonathanBieler) [inline src/GtkIDE.jl]: You can use `HOMEDIR` here instead of `dirname(@__FILE__)`.

**Comment by jonathanBieler** [comment]
There's still a few issues with this:
1. When I go to my Julia package directory, it freezes for maybe 10 seconds. I guess that reading all the directories recursively isn't the best idea :)
2. When it updates it folds all the tree, so you have to go and click again each time to view your files. There's probably a better way to update the list than rebuilding everything (or save and unfold the tree structure once it has been rebuilt).

**Comment by lucianolorenti** [comment]
Now the tree structure is loaded on demand and when a changes occur only the affected rows are updated

**Comment by jonathanBieler** [comment]
Cool, I think you need to do a new pull request with your new changes.

**Pull request merged**